.teamcity/CentralDeployment.kt

```diff
@@ -0,0 +1,214 @@
+        Platform.entries.forEach {
+            add(deployToCentral(it, startDeploymentTask))
+        }
+    }
+
+    deployTasks.forEach { deploy ->
+        copyToCentralTask.dependsOnSnapshot(deploy, onFailure = FailureAction.CANCEL)
+        copyZoneInfoTask.dependsOnSnapshot(deploy, onFailure = FailureAction.CANCEL)
+    }
+
+    buildTypesOrder = listOf(startDeploymentTask, *deployTasks.toTypedArray()) + listOf(copyToCentralTask, copyZoneInfoTask)
+}
+
+fun Project.startDeployment() = BuildType {
+    id("StartDeployment")
+    this.name = "Start Deployment [RUN THIS ONE]"
+    type = BuildTypeSettings.Type.DEPLOYMENT
+
+    params {
+        text(
+            "Version",
+            "",
+            display = ParameterDisplay.PROMPT,
+            allowEmpty = false
+        )
+        text(
+            "VersionSuffix",
+            "",
+            display = ParameterDisplay.PROMPT
+        )
+        select(
+            "Repository",
+            "central",
+            options = listOf("central", "sonatype"),
+            display = ParameterDisplay.PROMPT
+        )
+        text(
+            "ZoneInfoVersion",
+            "",
+            display = ParameterDisplay.HIDDEN
+        )
+    }
+
+    steps {
+        gradle {
+            name = "Get ZoneInfoVersion"
+            jdkHome = "%env.$jdk%"
+            jvmArgs = "-Xmx1g"
+            gradleParams =
+                "--info --stacktrace -P$versionSuffixParameter=%VersionSuffix% -P$releaseVersionParameter=%Version% -Pzoneinfo.version.tc.parameter=ZoneInfoVersion"
+            tasks = ":kotlinx-datetime-zoneinfo:setZoneInfoVersionToTeamcity"
+            buildFile = ""
+            gradleWrapperPath = ""
+        }
+    }
+
+    commonConfigure()
+}.also { buildType(it) }
+
+fun Project.deployToCentral(platform: Platform, startDeployment: BuildType) = buildType("DeployCentral", platform) {
+    type = BuildTypeSettings.Type.DEPLOYMENT
+    enablePersonalBuilds = false
+    maxRunningBuilds = 1
+    params {
+        param(versionSuffixParameter, "${startDeployment.depParamRefs["VersionSuffix"]}")
+        param(releaseVersionParameter, "${startDeployment.depParamRefs["Version"]}")
+        param("system.publication_repository", "${startDeployment.depParamRefs["Repository"]}")
+    }
+
+    vcs {
+        cleanCheckout = true
+    }
+
+    val taskNames = buildList {
+        add("clean")
+        when (platform) {
+            Platform.Linux -> {
+                addAll(
+                    listOf(
+                        "publishAndroidNativeArm32PublicationToCentralRepository",
+                        "publishAndroidNativeArm64PublicationToCentralRepository",
+                        "publishAndroidNativeX64PublicationToCentralRepository",
+                        "publishAndroidNativeX86PublicationToCentralRepository",
+                        "publishLinuxArm64PublicationToCentralRepository",
+                        "publishLinuxX64PublicationToCentralRepository"
```

Comment:
Where do all these tasks come from? `/gradlew tasks | grep publish | grep Central` returns nothing for me. The old `Deploy` builds only have `clean publish`, and I do have `publish` in my list of tasks.

Also, if this conditional publication depending on the platform is an optimization here, then can't `kotlinx-datetime` be optimized to just `clean publish` on just `MacOS`?

Reply:
> Where do all these tasks come from? 

You have to add following Gradle properties to get those tasks:
`-Pteamcity=on -Plibs.repo.url=https://localhost -Plibs.repo.user=user -Plibs.repo.password=password`

> Also, if this conditional publication depending on the platform is an optimization here, then can't `kotlinx-datetime` be optimized to just `clean publish` on just `MacOS`?

Sure, let's do that!

Reply:
Done